domain/src/main/scala/BitcoinCryptography.scala

```diff
@@ -0,0 +1,13 @@
+package org.parsec.protocol
+
+trait BitcoinCryptography {
```

Comment:
We could have something like:
````scala
/**
 * somewhere in the domain, and replacing the existing `string` types for keys in the protocol
 */
sealed trait PublicKey
case class BTCPublicKey(value: String) extends PublicKey {
  def p2pkhAddress = ???
  def p2shAddress = ???
  // plus validation methods...
}
case class ETHPublicKey(value: String) extends PublicKey {
  def address = ???
  // plus validation methods
}
````

That would let us keep all the methods with the data, and start moving us away from `String`s to value types (which will open up a lot more FP power as the project grows).